### PR TITLE
chore: rewrite imports in READMEs for aws-cdk-lib

### PR DESCRIPTION
**Pull request by njlynch**
For the v2 (aws-cdk-lib) docs, the imports scattered throughout the READMEs need
to be re-written. We could hand-edit all of the READMEs on the v2-main branch,
but this is not an ongoing mechanism and may also introduce forward-merge pain.
Instead, we're going to take the cheap option of rewriting the import statements
as part of `ubergen` when building the `aws-cdk-lib` package.

I aimed to capture as many import types as I found in the live READMEs.

As an example, the code transforms these imports as seen below:

```ts
// BEFORE
import * as cdk from '@aws-cdk/core';
  import * as cloudfront from '@aws-cdk/aws-cloudfront';
import codeartifact = require('@aws-cdk/aws-codeartifact');
import { VpcEndpointServiceDomainName } from '@aws-cdk/aws-route53';
import { Construct, Stage, Stack, StackProps, StageProps } from '@aws-cdk/core';

// AFTER
import * as cdk from 'aws-cdk-lib';
  import { aws_cloudfront as cloudfront } from 'aws-cdk-lib';
import { aws_codeartifact as codeartifact } from 'aws-cdk-lib';
import { VpcEndpointServiceDomainName } from 'aws-cdk-lib/aws-route53';
import { Construct, Stage, Stack, StackProps, StageProps } from 'aws-cdk-lib';
```

Note: Doesn't use the existing `rewrite-imports` method as that would require parsing the Markdown (requiring an extra dependency) and potentially guarding against malformed inputs. This approach was chosen as a simple, pragmatic solution. We can revisit this in the future if necessary.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: tools/ubergen/bin/ubergen.ts [hunk 1]
```diff
@@ -402,6 +408,37 @@ async function copyOrTransformFiles(from: string, to: string, libraries: readonl
...
+/**
+ * Rewrites the imports in README.md from v1 ('@aws-cdk/...') to v2 ('aws-cdk-lib').
+ * Uses the module imports (import { aws_foo as foo } from 'aws-cdk-lib') for module imports,
+ * and "barrel" imports for types (import { Bucket } from 'aws-cdk-lib/aws-s3').
+ */
+async function rewriteReadmeImports(fromFile: string): Promise<string> {
+  const readmeOriginal = await fs.readFile(fromFile, { encoding: 'utf8' });
+  return readmeOriginal
+    // import * as s3 from '@aws-cdk/aws-s3'
+    .replace(/^(\s*)import \* as (.*) from '@aws-cdk\/(.*)';(\s*)$/gm, rewriteCdkImports)
```
Comment (nija-at) [inline tools/ubergen/bin/ubergen.ts:420]: do we also need to take care of double quotes, besides single quotes?

**Review by nija-at — APPROVED** [review]

**Comment by rix0rrr** [comment]
PR could do with a motivation not to use the `rewrite-imports` functionality we already have (and presumably tested)

**Comment by nija-at** [comment]
> motivation not to use the `rewrite-imports` functionality

I was going to ask about this too. 
I then assumed it was because `rewrite-imports` used ts transforms that might not work on code snippets that are not valid typescript.
Would love to know if there were other reasons.

File: tools/ubergen/bin/ubergen.ts (hunk 1 above)
Comment (njlynch) [reply tools/ubergen/bin/ubergen.ts:420]: Good call :+1:. There are 14 imports across the entire codebase that use double-quotes. I'm tempted to fix them and leave the code as-is, but I suppose that's more fragile than just handling the quotes properly here.

**Pull request merged**